web/api/v1/api.go

```diff
@@ -1319,211 +1303,10 @@ func (api *API) serveTSDBStatus(*http.Request) apiFuncResult {
-	}
-
-	switch responseType {
-	case prompb.ReadRequest_STREAMED_XOR_CHUNKS:
-		api.remoteReadStreamedXORChunks(ctx, w, req, externalLabels, sortedExternalLabels)
-	default:
-		// On empty or unknown types in req.AcceptedResponseTypes we default to non streamed, raw samples response.
-		api.remoteReadSamples(ctx, w, req, externalLabels, sortedExternalLabels)
-	}
-}
-
-func (api *API) remoteReadSamples(
-	ctx context.Context,
-	w http.ResponseWriter,
-	req *prompb.ReadRequest,
-	externalLabels map[string]string,
-	sortedExternalLabels []prompb.Label,
-) {
-	w.Header().Set("Content-Type", "application/x-protobuf")
-	w.Header().Set("Content-Encoding", "snappy")
-
-	resp := prompb.ReadResponse{
-		Results: make([]*prompb.QueryResult, len(req.Queries)),
-	}
-	for i, query := range req.Queries {
-		if err := func() error {
-			filteredMatchers, err := filterExtLabelsFromMatchers(query.Matchers, externalLabels)
-			if err != nil {
-				return err
-			}
-
-			querier, err := api.Queryable.Querier(ctx, query.StartTimestampMs, query.EndTimestampMs)
-			if err != nil {
-				return err
-			}
-			defer func() {
-				if err := querier.Close(); err != nil {
-					level.Warn(api.logger).Log("msg", "Error on querier close", "err", err.Error())
-				}
-			}()
-
-			var hints *storage.SelectHints
-			if query.Hints != nil {
-				hints = &storage.SelectHints{
-					Start:    query.Hints.StartMs,
-					End:      query.Hints.EndMs,
-					Step:     query.Hints.StepMs,
-					Func:     query.Hints.Func,
-					Grouping: query.Hints.Grouping,
-					Range:    query.Hints.RangeMs,
-					By:       query.Hints.By,
-				}
-			}
-
-			var ws storage.Warnings
-			resp.Results[i], ws, err = remote.ToQueryResult(querier.Select(false, hints, filteredMatchers...), api.remoteReadSampleLimit)
-			if err != nil {
-				return err
-			}
-			for _, w := range ws {
-				level.Warn(api.logger).Log("msg", "Warnings on remote read query", "err", w.Error())
-			}
-			for _, ts := range resp.Results[i].Timeseries {
-				ts.Labels = remote.MergeLabels(ts.Labels, sortedExternalLabels)
-			}
-			return nil
-		}(); err != nil {
-			if httpErr, ok := err.(remote.HTTPError); ok {
-				http.Error(w, httpErr.Error(), httpErr.Status())
-				return
-			}
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
-		}
-	}
-
-	if err := remote.EncodeReadResponse(&resp, w); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
-	}
-}
-
-func (api *API) remoteReadStreamedXORChunks(ctx context.Context, w http.ResponseWriter, req *prompb.ReadRequest, externalLabels map[string]string, sortedExternalLabels []prompb.Label) {
-	w.Header().Set("Content-Type", "application/x-streamed-protobuf; proto=prometheus.ChunkedReadResponse")
-
-	f, ok := w.(http.Flusher)
-	if !ok {
-		http.Error(w, "internal http.ResponseWriter does not implement http.Flusher interface", http.StatusInternalServerError)
-		return
+	// This is only really for tests - this will never be nil IRL.
```

Comment:
```suggestion
	// This is only really for tests - this will never be nil IRL.
```

I'd not say that and properly return 404 for the importers.